presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloClient.java

```diff
@@ -0,0 +1,1140 @@
+import static com.facebook.presto.spi.StandardErrorCode.NOT_SUPPORTED;
+import static com.facebook.presto.spi.StandardErrorCode.USER_ERROR;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This class is the main access point for the Presto connector to interact with Accumulo. It is
+ * responsible for creating tables, dropping tables, retrieving table metadata, and getting the
+ * ConnectorSplits from a table.
+ * <p>
+ * Classes that requires an Accumulo Connector object should use the static method to retrieve it.
+ * This function will create a MiniAccumuloCluster connection for testing, or a 'real' one for production use.
+ */
+public class AccumuloClient
+{
+    private static final Logger LOG = Logger.get(AccumuloClient.class);
+    private static final String DUMMY_LOCATION = "localhost:9997";
+    private static final String MAC_PASSWORD = "secret";
+    private static final String MAC_USER = "root";
+
+    private static Connector conn = null;
+
+    /**
+     * Gets an Accumulo Connector based on the given configuration.
+     * Will start/stop MiniAccumuloCluster should that be true in the config.
+     *
+     * @param config Accumulo configuration
+     * @return Accumulo connector
+     */
+    public static synchronized Connector getAccumuloConnector(AccumuloConfig config)
+    {
+        if (conn != null) {
+            return conn;
+        }
+
+        try {
+            if (config.isMiniAccumuloCluster()) {
+                // Create MAC directory
+                final File macDir = Files.createTempDirectory("mac-").toFile();
+                LOG.info("MAC is enabled, starting MiniAccumuloCluster at %s", macDir);
+
+                // Start MAC and connect to it
+                MiniAccumuloCluster accumulo = new MiniAccumuloCluster(macDir, MAC_PASSWORD);
+                accumulo.start();
+                LOG.info("Connecting to: %s %s %s %s", accumulo.getInstanceName(),
+                        accumulo.getZooKeepers(), MAC_USER, MAC_PASSWORD);
+                Instance inst = new ZooKeeperInstance(accumulo.getInstanceName(), accumulo.getZooKeepers());
+                conn = inst.getConnector(MAC_USER, new PasswordToken(MAC_PASSWORD));
+                LOG.info("Connection established: %s %s %s %s", accumulo.getInstanceName(),
+                        accumulo.getZooKeepers(), MAC_USER, MAC_PASSWORD);
+
+                // Add shutdown hook to stop MAC and cleanup temporary files
+                Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+                    try {
+                        LOG.info("Shutting down MAC");
+                        accumulo.stop();
+                        LOG.info("Cleaning up MAC directory");
+                        FileUtils.forceDelete(macDir);
+                    }
+                    catch (IOException | InterruptedException e) {
+                        throw new RuntimeException(e);
```

Comment:
Made this into a `PrestoException`.  Will be making an `AccumuloErrorCode` class instead of using the `StandardErrorCode` in a future commit.